save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/SaveAgent.kt

```diff
@@ -121,6 +141,57 @@ class SaveAgent(private val config: AgentConfiguration,
         coroutineScope.cancel()
     }
 
+    // a temporary workaround for python integration
+    private fun executeAdditionallySetup(targetDirectory: Path, additionalFiles: List<FileKey>) = runCatching {
+        additionalFiles
+            .singleOrNull { it.name == "setup.sh" }
+            ?.let { fileKey ->
+                val targetFile = targetDirectory / fileKey.name
+                logDebugCustom("Additionally setup of evaluated tool by $targetFile")
+                val setupResult = ProcessBuilder(true, FileSystem.SYSTEM)
+                    .exec(
+                        "./$targetFile",
+                        "",
+                        null,
+                        SETUP_SH_TIMEOUT
+                    )
+                if (setupResult.code != 0) {
+                    throw IllegalStateException("${fileKey.name} is failed with error: ${setupResult.stderr}")
+                }
```

Comment:
Maybe also `logTrace` stdout if setup is successful for easier debugging?

Reply:
actually, we log info `logInfoCustom("Additionally setup has completed for execution $executionId")` as for rest actions

Reply:
Yes, but I mean for cases when, for example, for `pip install` from `setup.sh` we would like to see exactly which versions have been installed or something like this. IIRC, our `ProcessBuilder` captures stdout completely and doesn't replay it automatically.

Reply:
got it, will add logging of stdout